diff --git a/yt_dlp/options.py b/yt_dlp/options.py
index 76d401c..5dcfcb8 100644
--- a/yt_dlp/options.py
+++ b/yt_dlp/options.py
@@ -2,2 +2,3 @@
 import contextlib
+import json
 import optparse
@@ -294,2 +295,83 @@ def _dict_from_options_callback(
 
+    def _export_options_callback(option, opt_str, value, parser: _YoutubeDLOptionParser):
+        from .extractor import list_extractor_classes
+        from .extractor.generic import GenericIE # Importing GenericIE is currently slow since it imports YoutubeIE
+
+        extractors = []
+
+        for ie in list_extractor_classes():
+            extractors.append({
+                "name": ie.IE_NAME,
+                "description": ie.description(markdown=False),
+                "broken": not ie.working(),
+                "age_limit": ie.age_limit or None,
+            })
+
+        data = {
+            'option_groups': [],
+            "extractors": extractors,
+            'channel': CHANNEL,
+            'version': __version__,
+        }
+
+        for group in parser.option_groups:
+            group_data = {
+                'name': group.title,
+                'description': group.description,
+                'options': []
+            }
+            for option in group.option_list:
+                if option.dest == parser.ALIAS_DEST:
+                    continue
+
+                default = option.default
+
+                # if default isn't serializable, try to convert it to a type that is.
+                if default == optparse.NO_DEFAULT:
+                    default = None
+                elif not isinstance(default, (str, int, float, bool, list, dict, type(None))):
+                    try:
+                        default = str(default)
+                    except Exception:
+                        default = None
+
+                option_data = {
+                    'id': option.dest,
+                    'action': str(option.action),
+                    'choices': list(option.choices) if option.choices else None,
+                    'help': option.help if option.help != optparse.SUPPRESS_HELP else None,
+                    'hidden': option.help == optparse.SUPPRESS_HELP,
+                    'meta_args': option.metavar,
+                    'type': str(option.type) if option.type else None,
+                    'long_flags': option._long_opts,
+                    'short_flags': option._short_opts,
+                    'nargs': option.nargs if option.nargs and option.nargs > 0 and option.takes_value() else 0,
+                    'default_value': default,
+                    'const_value': option.const,
+                }
+
+                if not option_data['id'] or option_data['id'] == '_':
+                    if option._long_opts:
+                        option_data['id'] = option._long_opts[-1].lstrip('-')
+                    elif option.callback:
+                        option_data['id'] = option.callback.__name__.lstrip('_')
+
+                if not option_data['type']:
+                    if option.action == 'store_true' or option.action == 'store_false':
+                        option_data['type'] = 'bool'
+                    elif option.nargs and option.nargs > 0:
+                        option_data['type'] = 'string'
+                elif option_data['type'] == 'choice':
+                    option_data['type'] = 'string'
+
+                # if help output contains %default, replace it with the actual default value.
+                if option_data['help'] and '%default' in option_data['help']:
+                    option_data['help'] = option_data['help'].replace('%default', str(option_data['default_value']))
+
+                group_data['options'].append(option_data)
+            data['option_groups'].append(group_data)
+
+        print(json.dumps(data, indent=4))
+        sys.exit()
+
     def when_prefix(default):
@@ -352,2 +434,6 @@ def _preset_alias_callback(option, opt_str, value, parser):
     general = optparse.OptionGroup(parser, 'General Options')
+    general.add_option(
+        '--export-options',
+        action='callback', default=False, callback=_export_options_callback,
+        help=optparse.SUPPRESS_HELP)
     general.add_option(
